api/CcsSso.Core.Service/External/OrganisationProfileService.cs

```diff
@@ -687,13 +687,40 @@ private void Validate(OrganisationProfileInfo organisationProfileInfo)
           throw new CcsSsoException(ErrorConstant.ErrorInsufficientDetails);
         }
 
-        if (!string.IsNullOrWhiteSpace(organisationProfileInfo.Address.CountryCode) && !CultureSupport.IsValidCountryCode(organisationProfileInfo.Address.CountryCode))
+        string CountryName=String.Empty;
+        if (!string.IsNullOrEmpty(organisationProfileInfo.Address.CountryCode))
+        {
+          CountryName = GetCountryNameByCode(organisationProfileInfo.Address.CountryCode);
+        }
+        if (!string.IsNullOrWhiteSpace(organisationProfileInfo.Address.CountryCode) && string.IsNullOrWhiteSpace(CountryName))
         {
           throw new CcsSsoException(ErrorConstant.ErrorInvalidCountryCode);
         }
       }
     }
 
+    /// <summary>
+    /// Retrieves CountryName based on country code
+    /// </summary>
+    /// <returns></returns>
+    public string GetCountryNameByCode(string countyCode)
+    {
+      try
+      {
+        string CountryName = string.Empty;
+        CountryName = CultureSupport.GetCountryNameByCode(countyCode);
```

Comment:
What is the Country name for a country code like GB-ENG in this line? Will that throw any exception since CultureSupport.GetCountryNameByCode doesnt have any country code like that?